module-sys/SystemManager/CpuSentinel.cpp

```diff
@@ -1,18 +1,27 @@
-// Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+// Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
 // For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
 #include <SystemManager/CpuSentinel.hpp>
 #include "system/messages/RequestCpuFrequencyMessage.hpp"
 #include "system/Constants.hpp"
+#include <Timers/TimerFactory.hpp>
 #include <memory>
 
 namespace sys
 {
+    namespace
+    {
+        constexpr std::chrono::seconds defaultHoldFrequencyTime{1};
+    } // namespace
 
     CpuSentinel::CpuSentinel(std::string name,
                              sys::Service *service,
                              std::function<void(bsp::CpuFrequencyMHz)> callback)
-        : name(name), owner(service), callback(callback)
+        : name(name), owner(service),
+          callback(callback), timerHandle{sys::TimerFactory::createSingleShotTimer(
```

Comment:
Just asking. Wouldn't it be better to introduce a separate inheriting class like TimedCpuSentinel? Or there are some drawbacks with such approach here?

Reply:
You're right. A separate sentinel would be a good idea here

Reply:
done